acme: add DNS challenges

cjlapao opened the pull request:
<!-- _Please make sure to review and check all of these items, otherwise we might refuse your PR:_ -->

## Contribution Guidelines

* [x] I've read the [contribution guidelines](https://github.com/mailcow/mailcow-dockerized/blob/master/CONTRIBUTING.md) and wholeheartedly agree them

<!-- _NOTE: this tickbox is needed to fullfil on order to get your PR reviewed._ -->

## What does this PR include?

- New script for the acme.sh provider to support DNS-101 on acme-mailcow container #5435 

### Short Description

An implementation for the acme-mailcow docker to use the DNS-101 provider to get certificates, this will still allow the current flow as default and does not touch it, but also allows for the more modern and secure DNS-101 providers

###  Affected Containers

- acme-mailcow

## Did you run tests?

Yes but on a instalation as it is difficult to test otherwise, I also made the container build available here for testing. docker.io/cjlapao/mailcow-acme:1.0

### What did you tested?

- moved my current implementation of mailcow to use my docker image
- Added the new variables into the docker compose
- checked if certs were being issued
- removed the new environment variables and checked if the old mode still runs 

### What were the final results? (Awaited, got)

I got my certificates to run and work as expected

cjlapao committed:
Add DNS-01 challenge support for ACME certificates and related configurations

cjlapao committed:
Add DNS-01 challenge support with configuration files and scripts

DerLinkman commented:
Hi, to come back to  this PR:

The generate config part still contains the DNS_CHALLENGE Variables. And the update.sh or to be specific the _modules/core.sh does not include the update to add these Parameters to existing mailcow instances.

cjlapao committed:
Add DNS-01 challenge variables to update migration for existing instances

Existing mailcow installations were Not receiving the ACME_DNS_CHALLENGE,
ACME_DNS_PROVIDER, and ACME_ACCOUNT_EMAIL options when running update.sh,  as they were only defined in generate_config.sh for new installations.

magn2o commented:
I applied this PR locally and it's been working perfectly. I would just suggest a note indicating that all dns specific variables are stored in dns-101.conf, but other than that, no notes! 

The commented values in the mailcow.conf could be misleading:

> \# You will need to pass provider-specific environment variables to the acme-mailcow container.
> \# See the dns-101 provider documentation for more information.
> \# for example for Azure DNS:
> \#AZUREDNS_SUBSCRIPTIONID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> \#AZUREDNS_TENANTID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> \#AZUREDNS_APPID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> \#AZUREDNS_CLIENTSECRET=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> \#AZUREDNS_RESOURCEGROUP="your-resource-group"
> \#AZUREDNS_ZONE="your-zone-name"
> 

.. it makes you think those configuration values are stored there. 

Tested against both Namecheap and Cloudflare DNS.
